### PR TITLE
improve pull request template

### PR DESCRIPTION
**Pull request by grosser**
I usually just `Cmd+A - delete` since there is too much noise ... so I'd like to try and reduce noise here.
- start each sentence with a unique verb to make it easy to scan
- add changelog template note
- remove redundant words like "please make sure to" etc

/cc @estroz

### CODE REVIEW

**Comment by grosser** [comment]
markdown check failed but the output is unreadable 🤷 
https://github.com/operator-framework/operator-sdk/runs/689654457

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,19 +1,13 @@
 <!--
 
-Before making a PR, please read our contributing guidelines https://github.com/operator-framework/operator-sdk/blob/master/CONTRIBUTING.MD
-
-Note: Make sure your branch is rebased to the latest upstream master.
+- Read the contributing guidelines https://github.com/operator-framework/operator-sdk/blob/master/CONTRIBUTING.MD
```
Comment (estroz) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
Welcome to the Operator SDK! Before contributing, make sure to:

- Read the contributing guidelines https://github.com/operator-framework/operator-sdk/blob/master/CONTRIBUTING.MD
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,19 +1,13 @@
...
+- Rebase your branch on the latest upstream master
+- Add a changelog file by copying changelog/fragments/00-template.yaml 
+- When fixing an issue, add "Closes #<ISSUE_NUMBER>"
```
Comment (estroz) [inline .github/PULL_REQUEST_TEMPLATE.md:6]:
```suggestion
- Link any relevant issues, PR's, or documentation
- When fixing an issue, add "Closes #<ISSUE_NUMBER>"
```

**Comment by grosser** [comment]
updated

**Pull request merged**